### DISCUSSION TITLE
GPU doesn't always follow the targeted Fan Speed

### DISCUSSION THREAD

**Discussion by kayahr** [foucault/nvfancontrol#45] [General]
I'm trying to solve a problem where the GPU Fan gets extremely loud when GPU temperature reaches 80 degrees. So I tried nvfancontrol but it didn't help. 

I run `./nvfancontrol  -l 50,50` which should keep the fan constantly at 50%. This works at the beginning, I hear the constant fan speed, even nvidia-settings says the target speed is 50. But when I start a game then the fan speed is not constant anymore. And when temperature reaches 80 then the fan kicks in at full power.

I watched the settings GPUTargetFanSpeed and GPUCurrentFanSpeed and noticed that GPUTargetFanSpeed is always at 50 but the readonly setting GPUCurrentFanSpeed is 118 when temperature is at 80 degrees.

Even when system is idle then nvfancontrol seems to loose control over the fan. Target Speed and Current Speed drops to 0 after a while, then after some time it gets back to 50 as it should.

What the heck is going on here? I have no other Fan Control program running (at least none I know of and I'm pretty sure there is none by default). It feels like there is some automatic control coming from the NVidia driver itself which overrides the settings set by nvfancontrol. Maybe newer nvidia drivers need some more tweaking to disable the built-in fan control? 

System:
  * Debian 11
  * NVidia RTX 2080 Super
  * NVidia Driver 525.60.13
  * CoolBits 4

**Comment by foucault** [comment]
Hi! By default nvfancontrol won't touch the fans if they are already spinning in Auto mode (according to the card's firmware). I would suggest you add `-f` to force it to adjust speed even if the fans are already spinning.

**Reply by kayahr** [reply]
Didn't help. Fan still goes mad when temperature reaches 80 degrees.

I started nvfancontrol with parameters `-d -f -l 70,70`. It held 70% for a long time in idle mode. Then I started a game (Elite Dangerous on Steam via Proton), waited for the temperature to reach 80 and then the fan switches to full power.

The debug output might be interesting:

```
$ ./nvfancontrol -d -f -l 70,70
WARN - No config file found; using default curve
DEBUG - Default configuration loaded
DEBUG - [[gpu]]
id = 0
enabled = true
points = [[41, 20], [49, 30], [57, 45], [66, 55], [75, 63], [78, 72], [80, 80]]
DEBUG - Curve points: [(41, 20), (49, 30), (57, 45), (66, 55), (75, 63), (78, 72), (80, 80)]
INFO - NVIDIA driver version: 525.78.01
INFO - NVIDIA graphics adapter #0: NVIDIA GeForce RTX 2080 SUPER
INFO -   GPU #0 coolers: COOLER-0
...
DEBUG - Temp: 79; Speed: [2172] RPM ([70]%); Load: 50%; Mode: Manual
DEBUG - Temp: 79; Speed: [2168] RPM ([70]%); Load: 51%; Mode: Manual
DEBUG - Temp: 79; Speed: [2172] RPM ([70]%); Load: 51%; Mode: Manual
DEBUG - Temperature outside curve; setting to max
DEBUG - Temp: 80; Speed: [2541] RPM ([82]%); Load: 51%; Mode: Manual
DEBUG - Temp: 79; Speed: [3925] RPM ([127]%); Load: 51%; Mode: Manual
DEBUG - Temp: 79; Speed: [3925] RPM ([127]%); Load: 59%; Mode: Manual
```

This "Temperature outside curve; setting to max"  output was the moment the GPU reached 80 degrees.

Or is this normal behavior and 80 degrees is a bad thing for an RTX 2080 Super? According to nvidia-settings the GPU throttles down at 97 degrees. I think 80 degrees is a long distance away from 97 and this shouldn't force the fans to go to full-power mode.

**Reply by kayahr** [reply]
I tried again with a custom configuration where I added points for 90 and 100 degrees and set all levels to 70 for testing. Same problem, but this time no "setting to max" output:

```
$ ./nvfancontrol -d -f
INFO - Loading configuration file: "/home/k/.config/nvfancontrol.conf"
DEBUG - Curve points: [(30, 70), (40, 70), (50, 70), (60, 70), (70, 70), (90, 70), (100, 70)]
INFO - NVIDIA driver version: 525.78.01
INFO - NVIDIA graphics adapter #0: NVIDIA GeForce RTX 2080 SUPER
INFO -   GPU #0 coolers: COOLER-0
DEBUG - Temp: 50; Speed: [0] RPM ([0]%); Load: 5%; Mode: Manual
DEBUG - Temp: 50; Speed: [2620] RPM ([85]%); Load: 5%; Mode: Manual
DEBUG - Temp: 50; Speed: [2215] RPM ([71]%); Load: 5%; Mode: Manual
DEBUG - Temp: 49; Speed: [2163] RPM ([70]%); Load: 5%; Mode: Manual
...
DEBUG - Temp: 79; Speed: [2168] RPM ([70]%); Load: 51%; Mode: Manual
DEBUG - Temp: 79; Speed: [2168] RPM ([70]%); Load: 51%; Mode: Manual
DEBUG - Temp: 81; Speed: [2168] RPM ([70]%); Load: 52%; Mode: Manual
DEBUG - Temp: 79; Speed: [2172] RPM ([70]%); Load: 51%; Mode: Manual
DEBUG - Temp: 81; Speed: [2168] RPM ([70]%); Load: 52%; Mode: Manual
DEBUG - Temp: 80; Speed: [3818] RPM ([123]%); Load: 52%; Mode: Manual
DEBUG - Temp: 79; Speed: [3925] RPM ([127]%); Load: 52%; Mode: Manual
DEBUG - Temp: 80; Speed: [3911] RPM ([127]%); Load: 52%; Mode: Manual
DEBUG - Temp: 79; Speed: [1942] RPM ([63]%); Load: 56%; Mode: Manual
DEBUG - Temp: 80; Speed: [2156] RPM ([70]%); Load: 54%; Mode: Manual
DEBUG - Temp: 81; Speed: [2173] RPM ([70]%); Load: 54%; Mode: Manual

```

As you can see, temperature reached 81 degrees for the first time, the speed stayed at 70 but when it reached 81 again a few seconds later the fan goes again to full power (127%).

I have no idea what causes this. Is it possible that the GPU firmware interferes here and has some strange and unconfigurable emergency fan mode?
